src/cameras.jl

```diff
@@ -51,150 +103,34 @@ end
 
 Camera centered over all vehicles, does not change the zoom level.
 """
-struct SceneFollowCamera <: Camera end
-function update_camera!(rendermodel::RenderModel, camera::SceneFollowCamera, scene::Frame{E}) where {E<:Entity}
+struct SceneFollowCamera <: Camera
+    state::CameraState
+end
+SceneFollowCamera(;kwargs...) = SceneFollowCamera(CameraState(;kwargs...))
+function update_camera!(camera::SceneFollowCamera, scene::Frame{E}) where {E<:Entity}
     C = sum([posg(veh.state)[1:2] for veh in scene])/length(scene)  # center of mass
-    set_camera!(rendermodel, x=C[1], y=C[2])
+    set_camera!(camera.cs, x=C[1], y=C[2])
+    # should also add capabilities for adapting zoom level to make all vehicles of the scene fit
```

Comment:
There should be an option that allows the `SceneFollowCamera` to adapt its zoom level so it can be a better replacement for the `FitToContentCamera`

Reply:
agree but let's maybe keep it for another PR